src/connectionManager.ts

```diff
@@ -0,0 +1,74 @@
+export interface Context {
+  id: string;
+  resourceID: string;
+  resourceType: string;
+  resourceDisplay?: string;
+}
+
+export interface Connection {
+  id: string;
+  url: URL;
+  context: Context[];
+}
+
+export class ConnectionManager {
+  private connections: Connection[];
```

Comment:
I'm thinking this would be easier to implement operations on if it were a `Record<string, Connection>` where the `string` is the id. Thoughts?